src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ScheduledMetricCollectorsExecutor.java

```diff
@@ -94,11 +114,23 @@ public void run() {
             metricsCollectors.entrySet()) {
           if (entry.getValue() <= currentTime) {
             PerformanceAnalyzerMetricsCollector collector = entry.getKey();
+            if (excludedCollectors.contains(collector)) {
+              StatsCollector.instance().logException(StatExceptionCode.COLLECTORS_MUTED);
+              continue;
+            }
             metricsCollectors.put(collector, entry.getValue() + collector.getTimeInterval());
             if (!collector.inProgress()) {
               collector.setStartTime(currentTime);
               metricsCollectorsTP.execute(collector);
             } else {
+              if (collectorsTakingTooLong.contains(collector)) {
+                collectorsTakingTooLong.remove(collector);
```

Comment:
Don't we need to remove a long running collector in future, once it runs within time?

Reply:
Adding a log line here also might be useful for debugging

Reply:
I made a change to add states in the collector themselves and not maintain a queue. Let me know your thoughts after the change.